docs/building-deployer.md

```diff
@@ -0,0 +1,75 @@
+# Building Application Deployer
+
+## Overview
+
+Each application version is defined by one and only one **deployer image**. It
+carries references to other container images the application uses at runtime.
+Other than that, it does not need any other auxiliary data and, thus is
+_fully encapsulated_.
+
+A deployer image is a docker container image that serves the following purposes:
+
+- Its file system contains well known metadata files that define various aspects
+  of deploying the application. One important aspect is to drive the deployment
+  configuration UI form.
+- It can be executed as a standalone `Job`. Taking in user-supplied input
+  parameters, the job's pod will install all components of the application
+  and exits.
+
+In order to install the application components, the deployer image carries the
+full manifests of all k8s resources that need to be installed.
+
+## Building your deployer
+
+Decide how to craft your application k8s manifests.
+
+- [Helm](https://helm.sh): Ideal if you have existing charts and want to import
+  them. Helm also offers a powerful templating framework; the downside is a
+  higher learning curve. Follow instructions [here](building-deployer-helm.md).
+- Simple templating using `envsubst` (env var substitutions): Ideal if you are
+  starting from scratch and want a low learning curve to get your app running on
+  k8s. The downside is that templating capabilities are very limited.
+  Follow instructions [here](building-deployer-envsubst.md).
+
+## Publishing a new version
+
+A new application version is submitted by pushing the corresponding deployer image
+to the staging repo.
+
+### Tags
+
+The deployer image **must** carry the primary track ID as its docker tag.
+Marketplace uses the _latest_ image of that primary track ID tag when it looks
```

Comment:
Only one image at a time can have a given docker tag, right? So is the "_latest_ image" in this case equivalent to "the image that (currently) has the primary track ID tag"? (at the time that Marketplace "looks" for new versions)
Maybe it'd be good to avoid using the word "latest" here given that it is a special Docker tag.

Reply:
Agreed. I reworded it to:

```
Marketplace uses the image last tagged with that primary track ID tag when it looks
```